### ISSUE TITLE
Reproduced results are different from the paper 

### ISSUE THREAD

**Issue by lyl-7** [saic-mdal/lama#104]
Hi, thanks for your awesome work!
I am trying to reproduce the results of CelebA reported in the paper, but I get different results.

I download the pretrained model from [https://disk.yandex.ru/d/EgqaSnLohjuzAg/lama-celeba-hq/lama-fourie](https://disk.yandex.ru/d/EgqaSnLohjuzAg/lama-celeba-hq/lama-fourier) and put it to /experiments/lama-fourier-celeba, and  then follow the instructions in README.md
```
# Make shure you are in lama folder
cd lama
export TORCH_HOME=$(pwd) && export PYTHONPATH=.

# Download CelebA-HQ dataset
# Download data256x256.zip from https://drive.google.com/drive/folders/11Vz0fqHS2rXDb5pprgTjpD7S2BAJhi1P

# unzip & split into train/test/visualization & create config for it
bash fetch_data/celebahq_dataset_prepare.sh

# generate masks for test and visual_test at the end of epoch
bash fetch_data/celebahq_gen_masks.sh

# Infer model on thick/thin/medium masks in 256 and run evaluation like this:
python3 bin/predict.py \
model.path=$(pwd)/experiments/lama-fourier-celeba/ \
indir=$(pwd)/celeba-hq-dataset/visual_test_256/random_thick_256/ \
outdir=$(pwd)/inference/celeba_random_thick_256 model.checkpoint=best.ckpt

python3 bin/evaluate_predicts.py \
$(pwd)/configs/eval2_gpu.yaml \
$(pwd)/celeba-hq-dataset/visual_test_256/random_thick_256/ \
$(pwd)/inferencec/celeba_random_thick_256 \
$(pwd)/inference/celeba_random_thick_256_metrics.csv
```

I notice the hint  'Note that we do not fix a random seed, so the results will be slightly different each time' in README.md. Although I have tried different seed, the reproduced results are still different from the paper. Here are my results:
<img width="406" alt="image" src="https://user-images.githubusercontent.com/53854041/161924018-07f21c13-bb08-4db6-b2bb-1a0857584152.png">

Do you know why the results are different from the paper? Could you provide the testsets and masks you generated? 
Thanks a lot!

**Comment by windj007** [comment]
Hi! The metrics are different is because the data is different. Fixing a random seed now does not help because the random seed used during data generation is unknown. The main thing is that the relative performance of the models is approximately the same. From that point, there is nothing wrong with your numbers.

If you want to compare with LaMa or other baselines in your paper - then you probably should generate your own data and evaluate all the baselines over it. This is the only way to guarantee a fair comparison. Unfortunately, there is no single evalset established in the inpainting literature - everybody just generate new valsets and infer all the baselines (thanks to available pretrained models).

Publishing the exact testsets and masks is currently problematic, because we store them in PNG (so jpg compression does not affect metrics) and thus the valsets are pretty big (tens of gigabytes).

If you describe in more detail what you're trying to achieve, I can probably suggest some other ways to do it.

**Comment by lyl-7** [comment]
> Hi! The metrics are different is because the data is different. Fixing a random seed now does not help because the random seed used during data generation is unknown. The main thing is that the relative performance of the models is approximately the same. From that point, there is nothing wrong with your numbers.
> 
> If you want to compare with LaMa or other baselines in your paper - then you probably should generate your own data and evaluate all the baselines over it. This is the only way to guarantee a fair comparison. Unfortunately, there is no single evalset established in the inpainting literature - everybody just generate new valsets and infer all the baselines (thanks to available pretrained models).
> 
> Publishing the exact testsets and masks is currently problematic, because we store them in PNG (so jpg compression does not affect metrics) and thus the valsets are pretty big (tens of gigabytes).
> 
> If you describe in more detail what you're trying to achieve, I can probably suggest some other ways to do it.

Thanks for your kind reply! 
I am a beginner in inpainting. Your advice is very helpful, and I think I should not get stuck in the numbers.

**Comment by windj007** [comment]
Closed issue for now. If you have any further questions, feel free to reopen the issue or create a new one.

**Comment by Ellohiye** [comment]
> > Hi! The metrics are different is because the data is different. Fixing a random seed now does not help because the random seed used during data generation is unknown. The main thing is that the relative performance of the models is approximately the same. From that point, there is nothing wrong with your numbers.
> > If you want to compare with LaMa or other baselines in your paper - then you probably should generate your own data and evaluate all the baselines over it. This is the only way to guarantee a fair comparison. Unfortunately, there is no single evalset established in the inpainting literature - everybody just generate new valsets and infer all the baselines (thanks to available pretrained models).
> > Publishing the exact testsets and masks is currently problematic, because we store them in PNG (so jpg compression does not affect metrics) and thus the valsets are pretty big (tens of gigabytes).
> > If you describe in more detail what you're trying to achieve, I can probably suggest some other ways to do it.
> 
> Thanks for your kind reply! I am a beginner in inpainting. Your advice is very helpful, and I think I should not get stuck in the numbers.

Hi! Did you set the random seed when generating the test mask? Are seed0 and seed100 tested on different data or the same data? Can I communicate with you, my QQ is 3559640395, looking forward to your reply!

**Comment by leayz-888** [comment]
> Hi, thanks for your awesome work! I am trying to reproduce the results of CelebA reported in the paper, but I get different results.
> 
> I download the pretrained model from [https://disk.yandex.ru/d/EgqaSnLohjuzAg/lama-celeba-hq/lama-fourie](https://disk.yandex.ru/d/EgqaSnLohjuzAg/lama-celeba-hq/lama-fourier) and put it to /experiments/lama-fourier-celeba, and then follow the instructions in README.md
> 
> ```
> # Make shure you are in lama folder
> cd lama
> export TORCH_HOME=$(pwd) && export PYTHONPATH=.
> 
> # Download CelebA-HQ dataset
> # Download data256x256.zip from https://drive.google.com/drive/folders/11Vz0fqHS2rXDb5pprgTjpD7S2BAJhi1P
> 
> # unzip & split into train/test/visualization & create config for it
> bash fetch_data/celebahq_dataset_prepare.sh
> 
> # generate masks for test and visual_test at the end of epoch
> bash fetch_data/celebahq_gen_masks.sh
> 
> # Infer model on thick/thin/medium masks in 256 and run evaluation like this:
> python3 bin/predict.py \
> model.path=$(pwd)/experiments/lama-fourier-celeba/ \
> indir=$(pwd)/celeba-hq-dataset/visual_test_256/random_thick_256/ \
> outdir=$(pwd)/inference/celeba_random_thick_256 model.checkpoint=best.ckpt
> 
> python3 bin/evaluate_predicts.py \
> $(pwd)/configs/eval2_gpu.yaml \
> $(pwd)/celeba-hq-dataset/visual_test_256/random_thick_256/ \
> $(pwd)/inferencec/celeba_random_thick_256 \
> $(pwd)/inference/celeba_random_thick_256_metrics.csv
> ```
> 
> I notice the hint 'Note that we do not fix a random seed, so the results will be slightly different each time' in README.md. Although I have tried different seed, the reproduced results are still different from the paper. Here are my results: <img alt="image" width="406" src="https://user-images.githubusercontent.com/53854041/161924018-07f21c13-bb08-4db6-b2bb-1a0857584152.png">
> 
> Do you know why the results are different from the paper? Could you provide the testsets and masks you generated? Thanks a lot!



> Hi, thanks for your awesome work! I am trying to reproduce the results of CelebA reported in the paper, but I get different results.
> 
> I download the pretrained model from [https://disk.yandex.ru/d/EgqaSnLohjuzAg/lama-celeba-hq/lama-fourie](https://disk.yandex.ru/d/EgqaSnLohjuzAg/lama-celeba-hq/lama-fourier) and put it to /experiments/lama-fourier-celeba, and then follow the instructions in README.md
> 
> ```
> # Make shure you are in lama folder
> cd lama
> export TORCH_HOME=$(pwd) && export PYTHONPATH=.
> 
> # Download CelebA-HQ dataset
> # Download data256x256.zip from https://drive.google.com/drive/folders/11Vz0fqHS2rXDb5pprgTjpD7S2BAJhi1P
> 
> # unzip & split into train/test/visualization & create config for it
> bash fetch_data/celebahq_dataset_prepare.sh
> 
> # generate masks for test and visual_test at the end of epoch
> bash fetch_data/celebahq_gen_masks.sh
> 
> # Infer model on thick/thin/medium masks in 256 and run evaluation like this:
> python3 bin/predict.py \
> model.path=$(pwd)/experiments/lama-fourier-celeba/ \
> indir=$(pwd)/celeba-hq-dataset/visual_test_256/random_thick_256/ \
> outdir=$(pwd)/inference/celeba_random_thick_256 model.checkpoint=best.ckpt
> 
> python3 bin/evaluate_predicts.py \
> $(pwd)/configs/eval2_gpu.yaml \
> $(pwd)/celeba-hq-dataset/visual_test_256/random_thick_256/ \
> $(pwd)/inferencec/celeba_random_thick_256 \
> $(pwd)/inference/celeba_random_thick_256_metrics.csv
> ```
> 
> I notice the hint 'Note that we do not fix a random seed, so the results will be slightly different each time' in README.md. Although I have tried different seed, the reproduced results are still different from the paper. Here are my results: <img alt="image" width="406" src="https://user-images.githubusercontent.com/53854041/161924018-07f21c13-bb08-4db6-b2bb-1a0857584152.png">
> 
> Do you know why the results are different from the paper? Could you provide the testsets and masks you generated? Thanks a lot!

Hello, can you provide the lama-fourie pre-trained model? The link given by the author is invalid, thank you!
